src/models/case.js

```diff
@@ -43,6 +71,77 @@ class Case extends Model {
       },
     };
   }
+
+  static get relationMappings() {
+    return {
+      requests: {
+        relation: Model.HasManyRelation,
+        modelClass: Request,
+        join: {
+          from: 'case.id',
+          to: 'request.caseId',
+        },
+      },
+    };
+  }
+
+  async $beforeInsert() {
+    const lastInsertedCase = await Case.query()
+      .select('caseNumber')
+      .orderBy('caseNumber', 'desc')
+      .orderBy('createdAt', 'desc')
+      .limit(1);
+
+    this.caseNumber = getCaseNumber(lastInsertedCase[0].caseNumber);
+  }
+
+  $afterValidate(json) {
+    super.$afterValidate(json);
+    const cases = json;
+
+    // if referenceStatus is pending, casePendingReason cannot be empty
+    if (cases.referenceStatus === 'PENDING') {
+      if (
+        cases.casePendingReason === '' ||
+        cases.casePendingReason === null ||
+        cases.casePendingReason === undefined
+      ) {
+        throw new ValidationError({
+          message: 'Case pending reason required',
+        });
+      }
+    }
+
+    // validate amountRequested > 0 and amountGranted > 0 if there is input
+    if (Number.isNaN(cases.amountRequested)) {
+      throw new ValidationError({
+        message: 'Amount requested must be a number greater than 0',
+      });
+    }
+    if (Number.isNaN(cases.amountGranted)) {
```

Comment:
as above

Reply:
as the number is defined as a decimal, unsigned doesn't seem to work. Hence added this extra validation to check for positive number